From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: zenntet <airkraftvideos@gmail.com>
Date: Tue, 26 May 2026 18:36:55 -0500
Subject: [PATCH] final brand


diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index 00b64911b778d982ab42bc96d575ce9e5ed668f4..47c138ddd396119260450fbbca605f43c4fc6737 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -1952,7 +1952,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
 
     @DontObfuscate
     public String getServerModName() {
-        return "Liminal SMP (ShadowPaper)"; // Paper
+        return "LiminalPaper"; // Liminal SMP Brand
     }
 
     public SystemReport fillSystemReport(SystemReport systemReport) {
